src/precompile.jl

```diff
@@ -1,33 +1,29 @@
-    # These make a difference for Makie's TTFP and should help others too
-    # Total time for this to run is ~2s. To time it, disable:
-    #   - the `_precompile_()` call in StaticArrays.jl
-    #   - the jl_generating_output line above
-    # and then do
-    #     @time StaticArrays._precompile_()
-
-    for T in (Float32, Float64, Int)   # some common eltypes
-        for S = 2:4                    # some common sizes
-            L = S*S
-            @assert precompile(Tuple{typeof(*),SMatrix{S, S, T, L},SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{typeof(*),SMatrix{S, S, T, L},SVector{S, T}})
-            @assert precompile(Tuple{typeof(inv),SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{typeof(transpose),SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{typeof(_adjoint),Size{(S, S)},SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{Core.kwftype(typeof(minimum)),NamedTuple{(:dims,), Tuple{Int}},typeof(minimum),SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{Core.kwftype(typeof(maximum)),NamedTuple{(:dims,), Tuple{Int}},typeof(maximum),SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{typeof(getindex),SMatrix{S, S, T, L},SOneTo{S-1},SOneTo{S-1}})
+@setup_workload begin
+    @compile_workload begin
+        for T in (Float32, Float64, Int)   # some common eltypes
+            for S = 2:4                    # some common sizes
+                L = S*S
+                x = fill(0., SMatrix{S, S, T, L})
+                x * x
+                y = fill(0., SVector{S, T})
+                x * y
+                inv(x)
+                transpose(x)
+                _adjoint(Size(S, S), x)
+                # @assert precompile(Tuple{Core.kwftype(typeof(minimum)),NamedTuple{(:dims,), Tuple{Int}},typeof(minimum),SMatrix{S, S, T, L}})
+                # @assert precompile(Tuple{Core.kwftype(typeof(maximum)),NamedTuple{(:dims,), Tuple{Int}},typeof(maximum),SMatrix{S, S, T, L}})
+                getindex(x, SOneTo(S-1), SOneTo(S-1))
+            end
         end
-    end
 
-    # Some expensive generators
-    @assert precompile(Tuple{typeof(which(__broadcast,(Any,Size,Tuple{Vararg{Size}},Vararg{Any},)).generator.gen),Any,Any,Any,Any,Any,Any})
-    @assert precompile(Tuple{typeof(which(_zeros,(Size,Type{<:StaticArray},)).generator.gen),Any,Any,Any,Type,Any})
-    @assert precompile(Tuple{typeof(which(_mapfoldl,(Any,Any,Colon,Any,Size,Vararg{StaticArray},)).generator.gen),Any,Any,Any,Any,Any,Any,Any,Any})
+        # Some expensive generators
+        # @assert precompile(Tuple{typeof(which(__broadcast,(Any,Size,Tuple{Vararg{Size}},Vararg{Any},)).generator.gen),Any,Any,Any,Any,Any,Any})
+        # @assert precompile(Tuple{typeof(which(_zeros,(Size,Type{<:StaticArray},)).generator.gen),Any,Any,Any,Type,Any})
+        # @assert precompile(Tuple{typeof(which(_mapfoldl,(Any,Any,Colon,Any,Size,Vararg{StaticArray},)).generator.gen),Any,Any,Any,Any,Any,Any,Any,Any})
```

Comment:
```suggestion
y .* x .* y'
zeros(y)
zeros(x)
```
I _think_ the `dims`-based `maximum`/`minimum` will cover the `mapfoldl`, not sure though. No biggie if not.